test/gracefulstop_test.go

```diff
@@ -140,12 +140,10 @@ func (s) TestGracefulStop(t *testing.T) {
 
 	// Dial the server. This will cause a connection to be accepted. This will
 	// also unblock the Close method .
```

Comment:
This comment doesn't seem correct after the change. No connection will be made by NewClient. Please update the comments in the test based on when each event in the delayed listener is being triggered.